projects/transformers/trainer_mixins/distillation.py

```diff
@@ -86,16 +79,34 @@ def __init__(
                                 If None, no decay is applied. Defaults to None.
         """
 
-        super().__init__(**kwargs)
+        super().__init__(*args, **kwargs)
+
+        mixin_args = self.args.mixin_args
+
+        teacher_names_or_paths = mixin_args.get("teacher_model_names_or_paths", None)
+        teacher_model_cache_dir = mixin_args.get("teacher_model_cache_dir", None)
+        kd_ensemble_weights = mixin_args.get("kd_ensemble_weights", None)
+        kd_factor_init = mixin_args.get("kd_factor_init", 1.0)
+        kd_factor_end = mixin_args.get("kd_factor_end", 1.0)
+        kd_temperature_init = mixin_args.get("kd_temperature_init", 1.0)
+        kd_temperature_end = mixin_args.get("kd_temperature_end", 1.0)
 
         # Validate teacher models
         assert (
-            teacher_models is not None
-        ), "When using KD mixin, teacher_models must be defined"
+            isinstance(teacher_names_or_paths, list) and len(teacher_names_or_paths) > 0
+        ), "When using KD mixin, teacher_models_name_or_path must be defined"
+
+        teacher_models = []
+        for model_name_or_path in teacher_names_or_paths:
+            teacher_model = AutoModelForMaskedLM.from_pretrained(
```

Comment:
This doesn't include model_kwargs. I see that most of the arguments are irrelevant in our case, but are we sure config is redundant and can be removed?

Reply:
I checked some of the args, and I just checked the others. Here's the rundown: 
* `config` will be automatically instantiated through the `from_pretrained` command
* `cache_dir` is still helpful and is passable through `teacher_model_cache_dir`
* `revision` could be helpful, but so far we haven't across a scenario where it is
* `use_auth_token`, this is used for private models saved to HF, which we don't have

So we should be good to go.

Reply:
I was just worried about config, but if it is automatically instantiated then that makes it redundant.